index.js

```diff
@@ -12,36 +12,176 @@ var config = require('./lib/config');
+   */
+  _doesFileExistInCurrentProject: function(relativePath) {
+    relativePath = path.join('app', relativePath);
+
+    try {
+      fs.statSync(relativePath);
+      return true;
+    } catch (err) {
+      return this._doesTemplateFileExist(relativePath);
+    }
   },
-  _coverageIsEnabled: function() {
-    return process.env[this._config().coverageEnvVar];
+
+  /**
+   * Check if a file exists within the dummy app
+   * @param {String} relativePath - path to file within dummy app
+   * @returns {Boolean} whether or not the file exists within the dummy app
+   */
+  _doesFileExistInDummyApp: function(relativePath) {
+    relativePath = path.join('tests', 'dummy', 'app', relativePath);
+
+    try {
+      fs.statSync(relativePath);
+      return true;
+    } catch (err) {
+      return this._doesTemplateFileExist(relativePath);
+    }
+  },
+
+  /**
+   * Check if a template file exists within the current app/addon
+   * Note: Template files are already compiled into JavaScript files so we must
+   * check for the pre-compiled .hbs file
+   * @param {String} relativePath - path to file within current app/addon
+   * @returns {Boolean} whether or not the file exists within the current app/addon
+   */
+  _doesTemplateFileExist: function(relativePath) {
+    relativePath = relativePath.replace('.js', '.hbs');
+
+    try {
+      fs.statSync(relativePath);
+      return true;
+    } catch (err) {
+      return false;
+    }
   },
-  _config: function() {
+
+  /**
+   * Filter out files that come from other Ember addons and do not live within this app/addon
+   * @param {String} name - name of current app/addon
+   * @param {String} relativePath - relative path to file
+   * @returns {Boolean} whether or not to filter out file
+   */
+  _filterOutAddonFiles: function(name, relativePath) {
+    if(relativePath.indexOf(name + '/') === 0) {
+      relativePath = relativePath.replace(name + '/', '')
+    }
+
+    if (relativePath.indexOf('dummy/') === 0) {
+      relativePath = relativePath.replace('dummy/', '');
+    }
+
+    var fileExists = (
+      this._doesFileExistInDummyApp(relativePath) ||
+      this._doesFileExistInCurrentProject(relativePath)
+    );
+
+    return !fileExists;
+  },
+
+  /**
+   * Get project configuration
+   * @returns {Configuration} project configuration
+   */
+  _getConfig: function() {
     return config(this.project.root);
+  },
+
+  /**
+   * Get paths to exclude from coverage
+   * @returns {Array<String>} exclude paths
+   */
+  _getExcludes: function() {
+    var excludes = this._getConfig().excludes || [];
+    var name = this.parent.pkg.name;
+    excludes.push(this._filterOutAddonFiles.bind(this, name));
+    return excludes;
+  },
+
+  /**
+   * Determine whether or not coverage is enabled
+   * @returns {Boolean} whether or not coverage is enabled
+   */
+  _isCoverageEnabled: function() {
+    var value = process.env[this._getConfig().coverageEnvVar];
```

Comment:
Why not go with `return !!process.env[this._getConfig().coverageEnvVar];`?  What does the extra complexity here gain us?  Is this just to support setting `COVERAGE=false`?  If that is it, maybe something like this would be simpler:

```js
var value  = process.env[this._getConfig().coverageEnvVar];

return value && value !== 'false';
```

Or (if `true` is the magic value):

```js
return process.env[this._getConfig().coverageEnvVar] + '' === 'true';
```